File: Minchelin/app/src/main/java/com/example/minchelin/MainActivity.java
package com.example.minchelin;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.MenuItem;
import android.widget.Button;
import android.widget.ImageButton;

import com.google.android.gms.maps.internal.IMapFragmentDelegate;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {
    FragmentManager fragmentManager = getSupportFragmentManager();
    private home homeFragment = new home();
    private mapPage mapFragment = new mapPage();
    private userPage userFragment = new userPage();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FragmentTransaction transaction = fragmentManager.beginTransaction();
        transaction.add(R.id.frameLayout, homeFragment).commit();
        BottomNavigationView bottomNavigationView = findViewById(R.id.navigationView);
        bottomNavigationView.setOnNavigationItemSelectedListener(new ItemSelectedListener());
    }

    class ItemSelectedListener implements BottomNavigationView.OnNavigationItemSelectedListener {
        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            FragmentTransaction transaction = fragmentManager.beginTransaction();

            switch(item.getItemId())
            {
                case R.id.home:
                    transaction.replace(R.id.frameLayout, homeFragment).commitAllowingStateLoss();
                    break;

                case R.id.map:
                    transaction.replace(R.id.frameLayout, mapFragment).commitAllowingStateLoss();
                    break;
                case R.id.user:
                    transaction.replace(R.id.frameLayout, userFragment).commitAllowingStateLoss();
                    break;
            }
            return true;
        }
    }
}

File: DailyMoney/app/src/main/java/com/example/dailymoney/MainActivity.kt
package com.example.dailymoney

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentTransaction
import androidx.recyclerview.widget.LinearLayoutManager
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.fragment_home.*
import kotlinx.android.synthetic.main.fragment_spend_list.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val bottomNavigation: BottomNavigationView = findViewById(R.id.navigationView)
//        val categoryInput: ArrayList<String> = ArrayList()

        supportFragmentManager.beginTransaction().replace(R.id.frameLayout, home()).commit()

        bottomNavigation.setOnNavigationItemSelectedListener { item ->
            when(item.itemId) {
                R.id.home -> {
                    supportFragmentManager.beginTransaction().replace(R.id.frameLayout, home()).commit()
                }
                R.id.list -> {
                    supportFragmentManager.beginTransaction().replace(R.id.frameLayout, spendList()).commit()
                }
                R.id.user -> {
                    supportFragmentManager.beginTransaction().replace(R.id.frameLayout, user()).commit()
                }
            }

            true
        }

    }

}
